contracts/token/ERC20/utils/SafeERC20.sol

```diff
@@ -47,18 +47,28 @@ library SafeERC20 {
 
     function safeIncreaseAllowance(IERC20 token, address spender, uint256 value) internal {
         uint256 newAllowance = token.allowance(address(this), spender) + value;
-        _callOptionalReturn(token, abi.encodeWithSelector(token.approve.selector, spender, newAllowance));
+        forceApprove(token, spender, newAllowance);
     }
 
     function safeDecreaseAllowance(IERC20 token, address spender, uint256 value) internal {
         unchecked {
             uint256 oldAllowance = token.allowance(address(this), spender);
             require(oldAllowance >= value, "SafeERC20: decreased allowance below zero");
             uint256 newAllowance = oldAllowance - value;
-            _callOptionalReturn(token, abi.encodeWithSelector(token.approve.selector, spender, newAllowance));
+            forceApprove(token, spender, newAllowance);
         }
     }
 
+    function forceApprove(IERC20 token, address spender, uint256 value) internal {
+        // solidity has lazy evaluation
```

Comment:
"Short-circuit evaluation" is the right term here.

I think this deserves more comments explaining what's going on.